lang/cs/Org.Apache.REEF.Common/Io/ITempFileCreator.cs

```diff
@@ -0,0 +1,34 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.Common.Io
+{
+    /// <summary>
+    /// Provide interface for wraps temp file folder and temp file folder creation
+    /// </summary>
+    [DefaultImplementation(typeof(WorkingDirectoryTempFileCreator))]
+    public interface ITempFileCreator
+    {
+        string TempFileFolder { get; }
```

Comment:
Wouldn't it be better to have methods that create new temp file names for the user? Maybe with configurable prefix and suffix and we put a GUID or such in the middle, making sure the file doesn't already exist.